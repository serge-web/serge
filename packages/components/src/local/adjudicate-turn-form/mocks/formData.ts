import { AdjudicateTurnFormPopulate, AttributeTypes } from '@serge/custom-types'

const attributes: AttributeTypes = []

const formData: AdjudicateTurnFormPopulate = {
  contactId: 'C234',
  status: [
    {
      name: 'Fishing',
      mobile: false
    },
    {
      name: 'Moored',
      mobile: false
    },
    {
      name: 'Transiting',
      mobile: true
    }
  ],
  speed: [10, 20, 30],
  visibleTo: [
    {
      name: 'Blue Force',
      colour: '#69c',
      id: 'Blue'
    },
    {
      name: 'Red Force',
      colour: '#f00',
      id: 'Red'
    },
    {
      name: 'White Force',
      colour: '#fff',
      id: 'umpire'
    }
  ],
  condition: ['Working', 'Disabled', 'Immobile', 'Destroyed'],
  attributes: attributes
}

export default formData
