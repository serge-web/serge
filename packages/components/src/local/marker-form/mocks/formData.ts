import { PerceptionFormData } from '@serge/custom-types'

const formData: PerceptionFormData = {
  populate: {
    perceivedForces: [
      {
        name: 'Blue Force',
        colour: '#69c',
        id: 'Blue'
      },
      {
        name: 'Red Force',
        colour: '#f00',
        id: 'Red'
      },
      {
        name: 'White Force',
        colour: '#fff',
        id: 'Umpire'
      }
    ],
    perceivedTypes: [
      { uniqid: 'aa', name: 'Fishing vessel' },
      { uniqid: 'a1', name: 'torpedo' },
      { uniqid: 'a2', name: 'Boghammer' },
      { uniqid: 'a3', name: 'Frigate' }
    ]
  },
  values: {
    perceivedNameVal: 'Test',
    perceivedForceId: 'Blue Force',
    perceivedForceColor: '#69c',
    perceivedTypeId: 'a3',
    assetId: 'c00012',
    iconURL: 'frigate.svg'
  }
}

export default formData
