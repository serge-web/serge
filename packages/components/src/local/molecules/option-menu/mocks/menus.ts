import { faBath, faClone, faFileDownload, faTrash } from '@fortawesome/free-solid-svg-icons'

export default [
  {
    onClick: (): void => window.alert('Click on menu #1'),
    icon: faBath,
    title: 'Save clean copy of wargame (no messages)',
    label: 'Clean copy'
  },
  {
    onClick: (): void => window.alert('Click on menu #2'),
    icon: faClone,
    title: 'Duplicate wargame',
    label: 'Duplicate'
  },
  {
    onClick: (): void => window.alert('Click on menu #3'),
    icon: faFileDownload,
    title: 'Export wargame',
    label: 'Export'
  },
  {
    onClick: (): void => window.alert('Click on menu #4'),
    icon: faTrash,
    title: 'Delete wargame',
    label: 'Delete'
  }
]
