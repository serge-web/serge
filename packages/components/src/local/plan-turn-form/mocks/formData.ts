const formData = {
  populate: {
    status: [{
      name: 'Moored',
      mobile: false
    },
    {
      name: 'Transiting',
      mobile: true
    }],
    speed: [10, 20, 30]
  },
  values: {
    statusVal: {
      name: 'Transiting',
      mobile: true
    },
    speedVal: 10,
    turnsVal: 5,
    condition: 'Working'
  }
}

export default formData
