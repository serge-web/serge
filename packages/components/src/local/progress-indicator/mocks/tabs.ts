export const tabs = [
  {
    name: 'overview',
    complete: false
  },
  {
    name: 'forces',
    complete: false
  },
  {
    name: 'channels',
    complete: false
  }
]

export const tabsVariant = [
  {
    name: 'overview',
    complete: true
  },
  {
    name: 'forces',
    complete: false
  },
  {
    name: 'channels',
    complete: false
  }
]

export default tabs
