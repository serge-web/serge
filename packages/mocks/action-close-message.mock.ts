import { CLOSE_MESSAGE } from '@serge/config'
import { CloseMessageAction } from '@serge/custom-types'
import { GameMessagesMock } from '.'

const actionCloseMessage: CloseMessageAction = {
  type: CLOSE_MESSAGE,
  payload: {
    "channel": GameMessagesMock[0].details.channel,
    "message": GameMessagesMock[0]
  }
}

export default actionCloseMessage
