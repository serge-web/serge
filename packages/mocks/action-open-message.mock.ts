import { OPEN_MESSAGE } from '@serge/config'
import { OpenMessageAction } from '@serge/custom-types'
import { GameMessagesMock } from '.'

const actionOpenMessage: OpenMessageAction = {
  type: OPEN_MESSAGE,
  payload: {
    "channel": GameMessagesMock[0].details.channel,
    "message": {
      ...GameMessagesMock[0],
      isOpen: false
    }
  }
}

export default actionOpenMessage
