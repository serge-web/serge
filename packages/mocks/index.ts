/***
* This file is autogenerated, please do not edit it.
* To generate an updated version, please run `yarn genreg`
***/

export { default as GameMessagesMock } from './game-messages.mock'
export { default as GameMessagesMockRFI } from './game-messages-rfi.mock'
export { default as AdminMessagesMock } from './admin-messages.mock'
export { default as ChatMessagesMock } from './chat-messages.mock'
export { default as infoMarker } from './map-annotation.mock'
export { default as InfoMessagesMock } from './info-messages.mock'
export { default as annotationMarkerData } from './annotation-marker-data.mosk'
export { default as GameChannels2 } from './game-channels-2.mock'
export { default as WargameMock } from './wargame.mock'
export { default as MapAnnotationMock } from './map-annotation.mock'
export { default as WargameExportedMock } from './wargame-exported.mock'
export { default as P9Mock } from './p9.mock'
export { default as MessageTemplatesMock } from './message-templates.mock'
export { default as MessageTemplatesMockByKey } from './message-templates-by-key.mock'
export { default as localMappingConstraints } from './local-mapping.mock'
export { default as forces } from './forces.mock'
export { default as atlanticForces } from './atlantic-forces.mock'
export { default as smallForces } from './small-forces.mock'
export { default as smallScaleForces } from './small-scale-forces.mock'
export { default as watuWargame } from './watu-wargame-mock'
export { default as watuPlaytest } from './watu-playtest-mock'
export { default as p9wargame } from './p9.mock'
export { default as cmdWkWargame } from './cmd-week.mock'
export { default as smallScaleMappingConstraints } from './small-scale-mapping-constraints.mock'
export { default as preInitialisedForces } from './pre-initialised-forces.mock'
export { default as platformTypes } from './platform-types.mock'
export { default as platformType } from './platform-type.mock'
export { default as selectedAsset } from './selected-asset.mock'
export { default as adminTabs } from './admin-tabs.mock'
export { default as feedbackMessage, feedbackMessage1, feedbackMessage2 } from './feedback-message.mock'
export { default as actionFeedbackMessages } from './action-feedback-messages.mock'
export { default as actionSetLatestFeedbackMessage } from './action-set-latest-feedback-message.mock'
export { default as actionSetLatestWargameMessage } from './action-set-latest-wargame-message.mock'
export { default as actionSetWargameMessages } from './action-set-wargame-messages.mock'
export { default as actionOpenMessage } from './action-open-message.mock'
export { default as actionCloseMessage } from './action-close-message.mock'
export * from './coa-channel-data-2.mock'
export * from './planning-messages.mock'
export { default as planningMessageTemplatesMock } from './planning-message-templates.mock'
export * from './coa-messages.mock'
