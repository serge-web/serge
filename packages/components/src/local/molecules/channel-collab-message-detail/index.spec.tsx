/* global it expect */
import React from 'react'
import renderer from 'react-test-renderer'
import ChannelCollabMessageDetail from './index'

import { GameMessagesMockRFI } from '@serge/mocks'

const [defMessage] = GameMessagesMockRFI

describe('ChannelMessageDetail:', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ChannelCollabMessageDetail
        role={ { forceId: 'umpire', forceName: 'Umpire', roleId: 'co', roleName: 'CO' } }
        isUmpire={true}
        isRFIManager={true}
        message={defMessage}
        onChange={(msg): void => { console.log(msg) }}
      />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
