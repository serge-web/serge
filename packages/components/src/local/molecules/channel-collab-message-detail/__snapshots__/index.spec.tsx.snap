// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelMessageDetail: renders correctly 1`] = `
<div
  className="main"
>
  <div
    className="main theme-dark disabled"
  >
    <label
      htmlFor="question_2020-03-25T15:08:47.525Z"
    >
      Request
    </label>
    <textarea
      disabled={true}
      id="question_2020-03-25T15:08:47.525Z"
      onChange={[Function]}
      value="[message empty]"
    />
  </div>
  <div
    className="main theme-dark disabled"
  >
    <label
      htmlFor="answer_2020-03-25T15:08:47.525Z"
    >
      Answer
    </label>
    <textarea
      disabled={true}
      id="answer_2020-03-25T15:08:47.525Z"
      onChange={[Function]}
      value=""
    />
  </div>
  <div
    className="main theme-dark disabled"
  >
    <label
      htmlFor="private_message_2020-03-25T15:08:47.525Z"
    >
      <svg
        className="svg-inline--fa fa-user-secret"
      />
       
      Private Message
    </label>
    <textarea
      disabled={true}
      id="private_message_2020-03-25T15:08:47.525Z"
      onChange={[Function]}
      value=""
    />
  </div>
  <div
    className="actions"
  >
    <button
      className="MuiButtonBase-root MuiButton-root makeStyles-root-1 makeStyles-root-3 MuiButton-contained makeStyles-contained-2 makeStyles-contained-4 MuiButton-containedSizeSmall MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Take Ownership
      </span>
    </button>
  </div>
</div>
`;
